tests/misc/errors/js/smart-pipeline/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,61 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`snippet: #0 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> (yield y)}
+    |                                       ^"
+`;
+
+exports[`snippet: #1 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> (yield)}
+    |                                       ^"
+`;
+
+exports[`snippet: #2 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> (await y)}
+    |                                       ^"
+`;
+
+exports[`snippet: #3 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> (a?.b)}
+    |                                       ^"
+`;
+
+exports[`snippet: #4 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> a?.b}
+    |                                       ^"
+`;
+
+exports[`snippet: #5 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> (a.b())}
+    |                                       ^"
+`;
+
+exports[`snippet: #6 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> a.b()}
+    |                                       ^"
+`;
+
+exports[`snippet: #7 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> (a.b?.())}
+    |                                       ^"
+`;
+
+exports[`snippet: #8 [babel] format 1`] = `
+"Pipeline is in topic style but does not use topic reference (1:39)
+> 1 | async function * a() { a |> foo(#) |> a.b?.()}
+    |                                       ^"
+`;
+
+exports[`snippet: #9 [babel] format 1`] = `
+"Can not use 'yield' as identifier inside a generator (1:39)
+> 1 | async function * a() { a |> foo(#) |> yield}
```

Comment:
Not directly related to this PR, but if we remove the generator:
```js
a |> foo(#) |> yield
```
we'll get `Unexpected reserved word 'yield'` ([playground](https://deploy-preview-10446--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBDABAHwHzoGYQQAUAxAJRa4CeAlnADYAmIANCBAA4w3QDOyoVACchEAO4AFYQn4pU9Mair82AIyGowAazgwAyh000oAc2QwhAVzhs4AW1VxGjJwBlUpy6hNwAYhCE7VBhuU2Q0SxgIVhAACxg7egB1WJp4XkMwOD0ZNJoANzSqcLBeFRBjXjghGAkNEyDkPHkqtgArXgAPACENbV09VDs4V2M4JpabEA7OvWMTejgARUsIeAn6VpBDISqhcNVUR3oYjiFjGCSaRhhY5AAOAAY2M4gqpI0OcLO4PfzxtgAR1W8DqnFkaF4AFooHAnE4YkI4MCaEi6t5GkhmpsplU7DRzFZcfNFis1uMsZM2DAjlcbnckAAmakaGj0eYAYQgdkxIF+AFYYpYqgAVI6ybFbfLWACSUBcsD0YHOXAAgvK9DAqIsNlUAL56oA)), which should only happen in strict mode. In non-strict mode, `yield` is an acceptable identifier. Version 2.2.1 handles this better thanks to the error recovery.